packages/main/src/components/DynamicPage/index.tsx

```diff
@@ -0,0 +1,227 @@
+import { createComponentStyles } from '@ui5/webcomponents-react-base/lib/createComponentStyles';
+import { useConsolidatedRef, usePassThroughHtmlProps } from '@ui5/webcomponents-react-base/lib/hooks';
+import { StyleClassHelper } from '@ui5/webcomponents-react-base/lib/StyleClassHelper';
+import { ThemingParameters } from '@ui5/webcomponents-react-base/lib/ThemingParameters';
+import { CommonProps } from '@ui5/webcomponents-react/interfaces/CommonProps';
+import { FlexBox } from '@ui5/webcomponents-react/lib/FlexBox';
+import { GlobalStyleClasses } from '@ui5/webcomponents-react/lib/GlobalStyleClasses';
+import { PageBackgroundDesign } from '@ui5/webcomponents-react/lib/PageBackgroundDesign';
+import React, {
+  forwardRef,
+  ReactElement,
+  ReactNode,
+  ReactNodeArray,
+  Ref,
+  FC,
+  cloneElement,
+  RefObject,
+  useRef,
+  useCallback,
+  useState,
+  useEffect
+} from 'react';
+import { DynamicPageAnchorBar } from '../DynamicPageAnchorBar';
+import { useObserveHeights } from '../ObjectPage/useObserveHeights';
+import styles from './DynamicPage.jss';
+
+export interface DynamicPageProps extends CommonProps {
+  /**
+   * Determines the background color of DynamicPage.
+   */
+  backgroundDesign?: PageBackgroundDesign;
+  /**
+   * Determines whether the footer is visible.
+   */
+  showFooter?: boolean;
+  /**
+   * Determines whether the header is shown.
+   */
+  noHeader?: boolean;
+  /**
+   * Determines whether the content header is shown.
+   */
+  alwaysShowContentHeader?: boolean;
+  /**
+   * Determines whether the header button is shown.
+   */
+  showHideHeaderButton?: boolean;
+  /**
+   * Determines whether the pin button is shown.
+   */
+  headerContentPinnable?: boolean;
+
+  // slots
+  title?: ReactElement;
+
+  header?: ReactElement;
+
+  anchorBar?: ReactElement;
+
+  contentArea?: ReactElement;
```

Comment:
I think we should remove the `contentArea` and only use `children`. Or is there any difference?